react/src/pages/addParticipants/DataEntryTable.tsx

```diff
@@ -60,19 +60,55 @@ const useTableStyles = makeStyles(theme => ({
     },
 }));
 
-const defaultOptionals = ["notes", "sex", "input_hpf_path"];
+const fallbackColumns = ["notes", "sex", "input_hpf_path"];
+
+function getEnvColumns(): Array<keyof DataEntryRowOptional> | null {
+    const envCols = process.env.REACT_APP_DEFAULT_OPTIONAL_COLUMNS;
+    console.log(envCols);
+    if (envCols !== undefined) {
+        const validFields = getDataEntryHeaders().optional;
+        const envFields = envCols.split(",");
+        const usableFields = envFields.filter(
+            field => !!validFields.find(valid => valid === field)
+        );
+        return usableFields as Array<keyof DataEntryRowOptional>;
+    } else {
+        return null;
+    }
+}
+
+function getDefaultColumns(fallbackColumns: string[]) {
+    const storedDefaults = window.localStorage.getItem("data-entry-default-columns");
+    let tempCols = fallbackColumns;
+    const envCols = getEnvColumns();
+
+    if (storedDefaults !== null) {
+        // User already has stored preferences
+        tempCols = JSON.parse(storedDefaults);
+    } else if (Array.isArray(envCols) && envCols.length > 0) {
+        // No preferences, use .env
+        tempCols = envCols;
+    } else {
+        // No .env, use fallback columns
+        window.localStorage.setItem("data-entry-default-columns", JSON.stringify(tempCols));
+    }
+    return tempCols;
+}
 
 export default function DataEntryTable(props: DataEntryTableProps) {
     const classes = useTableStyles();
 
     const columns = getColumns("required");
     const RNASeqCols = getColumns("RNASeq");
 
-    const [optionals, setOptionals] = useState<DataEntryHeader[]>(
-        getColumns("optional").map(header => {
-            return { ...header, hidden: !defaultOptionals.includes(header.field) };
-        })
-    );
+    function getOptionalHeaders() {
+        return getColumns("optional").map(header => ({
+            ...header,
+            hidden: !getDefaultColumns(fallbackColumns).includes(header.field),
```

Comment:
`getDefaultColumns` gets called every time, which means you are reparsing the same values and saving the same thing to localStorage for every optional column.